packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/DocumentController.java

```diff
@@ -516,7 +519,9 @@ public ResponseEntity<byte[]> downloadDocument(
 	) {
 		try {
 			final Optional<byte[]> bytes = documentAssetService.fetchFileAsBytes(id, filename);
-			return bytes.map(ResponseEntity::ok).orElse(null);
+
+			final CacheControl cacheControl = CacheControl.maxAge(24, TimeUnit.HOURS).cachePublic();
+			return ResponseEntity.ok().cacheControl(cacheControl).body(bytes.orElse(null));
```

Comment:
Should we cache something that could be null? Artifact and Simulation controllers do not respond with OK if it's null.